source/fe/mapping_fe.cc

```diff
@@ -480,6 +480,7 @@ MappingFE<dim,spacedim,DH,VECTOR>::fill_fe_values (
                 else
                   {
                     const unsigned int codim = spacedim-dim;
+                    (void)codim;
 
                     if (update_flags & update_normal_vectors)
                       {
```

Comment:
The code where `codim` is used looks like this in the very next line:
```
  Assert( codim==1 , ExcMessage("There is no cell normal in codim 2."));
```
I think it would be easier to just replace this by
```
  Assert( spacedim-dim==1 , ExcMessage("There is no cell normal in codimensions greater than one."));
```
This way you don't need to offending variable at all.